### PR TITLE
[Merged by Bors] - feat: the space of continuous maps is complete if the target is complete

### PR DESCRIPTION
**Pull request by ocfnash**
---

Conversation [on Zulip](https://leanprover.zulipchat.com/#narrow/stream/116395-maths/topic/Completeness.20and.20compact.20convergence/near/464829091)

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/RestrictGenTopology.lean [hunk 1]
```diff
@@ -91,4 +91,14 @@ lemma of_seq [SequentialSpace X]
...
 
+lemma isCompact_of_weaklyLocallyCompact [WeaklyLocallyCompactSpace X] :
+    RestrictGenTopology {K : Set X | IsCompact K} := by
+  refine ⟨fun u hu ↦ isOpen_iff_mem_nhds.mpr fun x hx ↦ ?_⟩
+  obtain ⟨K, h₁, h₂⟩ := exists_compact_mem_nhds x
+  obtain ⟨v, h₃, h₄⟩ := inducing_subtype_val.isOpen_iff.mp <| hu K h₁
+  rw [Subtype.preimage_val_eq_preimage_val_iff] at h₄
+  have : K ∩ v ∈ 𝓝 x :=
+    inter_mem h₂ (h₃.mem_nhds <| mem_of_mem_inter_right (a := K) <| h₄ ▸ ⟨mem_of_mem_nhds h₂, hx⟩)
+  simp_all only [mem_setOf_eq, inter_mem_iff, and_true]
```
Comment (urkud) [inline Mathlib/Topology/RestrictGenTopology.lean:102]:
```suggestion
/-- If each point of the space has a neighborhood from the family `S`,
then the topology is generated by its restrictions to the sets of `S`. -/
lemma of_nhds (h : ∀ x, ∃ s ∈ S, s ∈ 𝓝 x) : RestrictGenTopology S :=
  of_continuous_prop fun _f hf ↦ continuous_iff_continuousAt.2 fun x ↦
    let ⟨s, hsS, hsx⟩ := h x
    (hf s hsS).continuousAt hsx

/-- A weakly locally compact space is compactly generated. -/
lemma isCompact_of_weaklyLocallyCompact [WeaklyLocallyCompactSpace X] :
    RestrictGenTopology {K : Set X | IsCompact K} :=
  of_nhds exists_compact_mem_nhds
```

File: Mathlib/Topology/UniformSpace/CompactConvergence.lean [hunk 2]
```diff
@@ -356,4 +360,24 @@ theorem uniformSpace_eq_iInf_precomp_of_cover {δ : ι → Type*} [∀ i, Topolo
     UniformSpace.comap_iInf, ← UniformSpace.comap_comap]
   rfl
 
+section CompleteSpace
+
+variable [CompleteSpace β]
+
+lemma completeSpace_of_restrictGenTopology (h : RestrictGenTopology {K : Set α | IsCompact K}) :
```
Comment (urkud) [inline Mathlib/Topology/UniformSpace/CompactConvergence.lean:367]: Please add a docstring.

**Comment by urkud** [comment]
Otherwise LGTM. Thanks!
bors d+

File: Mathlib/Topology/RestrictGenTopology.lean (hunk 1 above)
Comment (ocfnash) [reply Mathlib/Topology/RestrictGenTopology.lean]: Neat!

**Comment by ocfnash** [comment]
bors merge

**Pull request closed without merging**